yotta python setup.py test has errors.

```
simhug01@E103032 /c/develop/public/jobs/yr2015/2209/work/20150320/yotta (master)
$ python setup.py test
running test
running egg_info
creating yotta.egg-info
writing requirements to yotta.egg-info\requires.txt
writing yotta.egg-info\PKG-INFO
writing top-level names to yotta.egg-info\top_level.txt
writing dependency_links to yotta.egg-info\dependency_links.txt
writing entry points to yotta.egg-info\entry_points.txt
writing manifest file 'yotta.egg-info\SOURCES.txt'
reading manifest file 'yotta.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'yotta.egg-info\SOURCES.txt'
running build_ext
test_installDeps (yotta.test.cli.install.TestCLIInstall) ... skipped 'a github authtoken must be specified for this test
 (run yotta login, or set YOTTA_GITHUB_AUTHTOKEN)'
test_installGithubRef (yotta.test.cli.install.TestCLIInstall) ... skipped 'a github authtoken must be specified for this
 test (run yotta login, or set YOTTA_GITHUB_AUTHTOKEN)'
test_installRegistryRef (yotta.test.cli.install.TestCLIInstall) ... skipped 'a github authtoken must be specified for th
is test (run yotta login, or set YOTTA_GITHUB_AUTHTOKEN)'
test_publishPrivate (yotta.test.cli.publish.TestCLIPublish) ... FAIL
test_both (yotta.test.cli.search.TestCLISearch) ... ok
test_bothModule (yotta.test.cli.search.TestCLISearch) ... ok
test_bothTarget (yotta.test.cli.search.TestCLISearch) ... ok
test_keywords (yotta.test.cli.search.TestCLISearch) ... ok
test_limit (yotta.test.cli.search.TestCLISearch) ... ok
test_modules (yotta.test.cli.search.TestCLISearch) ... ok
test_targets (yotta.test.cli.search.TestCLISearch) ... ok
test_setTarget (yotta.test.cli.target.TestCLITarget) ... ok
test_setTargetLocal (yotta.test.cli.target.TestCLITarget) ... ok
test_bumpVersion (yotta.test.cli.version.TestCLIVersion) ... ok
test_displayVersion (yotta.test.cli.version.TestCLIVersion) ... ok
test_component_init (yotta.test.components.ComponentTestCase) ... ok
test_branchAccess (yotta.test.github_access.TestGitHubAccess) ... skipped 'a github authtoken must be specified for this
 test (run yotta login, or set YOTTA_GITHUB_AUTHTOKEN)'
test_installDeps (yotta.test.github_access.TestGitHubAccess) ... skipped 'a github authtoken must be specified for this
test (run yotta login, or set YOTTA_GITHUB_AUTHTOKEN)'
test_availableVersions (yotta.test.git_access.TestGitAccess) ... ERROR
test_installDeps (yotta.test.git_access.TestGitAccess) ... The authenticity of host 'bitbucket.org (131.103.20.167)' can
't be established.
RSA key fingerprint is 97:8c:1b:f2:6f:14:6b:5c:3b:ec:aa:46:46:74:7c:40.
Are you sure you want to continue connecting (yes/no)? yes
ERROR:root:Failed to satisfy dependencies of c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\git-access-testing:
ERROR:root:command failed: ['git', 'clone', u'git@bitbucket.org:autopulated/testing-dummy.git', 'c:\\users\\simhug01\\ap
pdata\\local\\temp\\tmpeloy0c']:Cloning into 'c:\users\simhug01\appdata\local\temp\tmpeloy0c'...
Warning: Permanently added 'bitbucket.org,131.103.20.167' (RSA) to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

ERROR:root:command failed: ['git', 'clone', u'git@bitbucket.org:autopulated/other-testing-dummy.git', 'c:\\users\\simhug
01\\appdata\\local\\temp\\tmps4ggyo']:Cloning into 'c:\users\simhug01\appdata\local\temp\tmps4ggyo'...
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

ERROR
test_versionSpec (yotta.test.git_access.TestGitAccess) ... ERROR
test_availableVersions (yotta.test.hg_access.TestHGAccess) ... ERROR
test_installDeps (yotta.test.hg_access.TestHGAccess) ... ERROR
test_versionSpec (yotta.test.hg_access.TestHGAccess) ... ERROR
test_merging (yotta.test.settings.TestSettings) ... ok
test_setting (yotta.test.settings.TestSettings) ... ok
test_writing (yotta.test.settings.TestSettings) ... ok
test_gitURLs (yotta.test.sourceparse.TestGitAccess) ... ok
test_githubURLs (yotta.test.sourceparse.TestGitAccess) ... ok
test_hgURLs (yotta.test.sourceparse.TestGitAccess) ... ok
test_registryURLs (yotta.test.sourceparse.TestGitAccess) ... ok
test_componentNameCoerced (yotta.test.validation.TestValidation) ... ok
test_looksLikeAnEmail (yotta.test.validation.TestValidation) ... ok
test_validateSourceDirNames (yotta.test.validation.TestValidation) ... ok
test_validateSourceDirSuggestions (yotta.test.validation.TestValidation) ... ok
test_commit (yotta.test.vcs.TestGit) ... ERROR
test_isClean (yotta.test.vcs.TestGit) ... ERROR
test_commit (yotta.test.vcs.TestHg) ... ERROR
test_isClean (yotta.test.vcs.TestHg) ... ERROR
test_hash (yotta.test.versions.VersionTestCase) ... ok
test_matches (yotta.test.versions.VersionTestCase) ... ok

======================================================================
ERROR: test_availableVersions (yotta.test.git_access.TestGitAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\git_access.py", line 60, in tearDown
    fsutils.rmRf(self.working_copy.directory)
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\fsutils.py", line 28, in rmRf
    shutil.rmtree(path)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\mbed_tools\Python27\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\simhug01\\appdata\\local\\temp\\tmp7xmkjs\\.git\\objects\\pack\\pa
ck-96f465387151fc78996d62f90c6e08b30d6cbabb.idx'

======================================================================
ERROR: test_installDeps (yotta.test.git_access.TestGitAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\git_access.py", line 60, in tearDown
    fsutils.rmRf(self.working_copy.directory)
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\fsutils.py", line 28, in rmRf
    shutil.rmtree(path)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\mbed_tools\Python27\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\simhug01\\appdata\\local\\temp\\tmpiebhzw\\.git\\objects\\pack\\pa
ck-96f465387151fc78996d62f90c6e08b30d6cbabb.idx'

======================================================================
ERROR: test_versionSpec (yotta.test.git_access.TestGitAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\git_access.py", line 60, in tearDown
    fsutils.rmRf(self.working_copy.directory)
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\fsutils.py", line 28, in rmRf
    shutil.rmtree(path)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\mbed_tools\Python27\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\simhug01\\appdata\\local\\temp\\tmpnwqnbp\\.git\\objects\\pack\\pa
ck-96f465387151fc78996d62f90c6e08b30d6cbabb.idx'

======================================================================
ERROR: test_availableVersions (yotta.test.hg_access.TestHGAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 47, in setUp
    ensureHGConfig()
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 39, in ensureHGConfig
    info = hgapi.Repo.command(".", os.environ, "showconfig")
  File "c:\mbed_tools\Python27\lib\site-packages\hgapi\hgapi.py", line 89, in command
    stdout=PIPE, stderr=PIPE, env=env)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

======================================================================
ERROR: test_installDeps (yotta.test.hg_access.TestHGAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 47, in setUp
    ensureHGConfig()
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 39, in ensureHGConfig
    info = hgapi.Repo.command(".", os.environ, "showconfig")
  File "c:\mbed_tools\Python27\lib\site-packages\hgapi\hgapi.py", line 89, in command
    stdout=PIPE, stderr=PIPE, env=env)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

======================================================================
ERROR: test_versionSpec (yotta.test.hg_access.TestHGAccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 47, in setUp
    ensureHGConfig()
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\hg_access.py", line 39, in ensureHGConfig
    info = hgapi.Repo.command(".", os.environ, "showconfig")
  File "c:\mbed_tools\Python27\lib\site-packages\hgapi\hgapi.py", line 89, in command
    stdout=PIPE, stderr=PIPE, env=env)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

======================================================================
ERROR: test_commit (yotta.test.vcs.TestGit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\vcs.py", line 43, in tearDown
    self.working_copy.remove()
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\vcs.py", line 107, in remove
    fsutils.rmRf(self.worktree)
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\fsutils.py", line 28, in rmRf
    shutil.rmtree(path)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\mbed_tools\Python27\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\simhug01\\appdata\\local\\temp\\tmpsq7hqx\\.git\\objects\\1d\\51b4
50e4aa777a43c33a626ee2bc07a0922e37'

======================================================================
ERROR: test_isClean (yotta.test.vcs.TestGit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\vcs.py", line 43, in tearDown
    self.working_copy.remove()
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\vcs.py", line 107, in remove
    fsutils.rmRf(self.worktree)
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\lib\fsutils.py", line 28, in rmRf
    shutil.rmtree(path)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\mbed_tools\Python27\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\mbed_tools\Python27\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\simhug01\\appdata\\local\\temp\\tmpt4afaf\\.git\\objects\\pack\\pa
ck-96f465387151fc78996d62f90c6e08b30d6cbabb.idx'

======================================================================
ERROR: test_commit (yotta.test.vcs.TestHg)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\vcs.py", line 61, in setUp
    info = hgapi.Repo.command(".", os.environ, "showconfig")
  File "c:\mbed_tools\Python27\lib\site-packages\hgapi\hgapi.py", line 89, in command
    stdout=PIPE, stderr=PIPE, env=env)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

======================================================================
ERROR: test_isClean (yotta.test.vcs.TestHg)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\vcs.py", line 61, in setUp
    info = hgapi.Repo.command(".", os.environ, "showconfig")
  File "c:\mbed_tools\Python27\lib\site-packages\hgapi\hgapi.py", line 89, in command
    stdout=PIPE, stderr=PIPE, env=env)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\mbed_tools\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

======================================================================
FAIL: test_publishPrivate (yotta.test.cli.publish.TestCLIPublish)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\develop\public\jobs\yr2015\2209\work\20150320\yotta\yotta\test\cli\publish.py", line 67, in test_publishPriva
te
    self.assertTrue('is private and cannot be published' in ('%s %s' % (stdout, stderr)))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 41 tests in 48.891s

FAILED (failures=1, errors=10, skipped=5)
←[0m
simhug01@E103032 /c/develop/public/jobs/yr2015/2209/work/20150320/yotta (master)
$
```

--- a/yotta/lib/fsutils.py
+++ b/yotta/lib/fsutils.py
@@ -8,6 +8,7 @@ import os
 import errno
 import shutil
 import platform
+import stat
 
 def mkDirP(path):
     try:
@@ -24,8 +25,16 @@ def rmF(path):
             raise
 
 def rmRf(path):
+    # we may have to make files writable before we can successfully delete
+    # them, to do this
+    def fixPermissions(fn, path, excinfo):
+        if os.access(path, os.W_OK):
+            raise
+        else:
+            os.chmod(path, stat.S_IWUSR)
+            fn(path)
     try:
-        shutil.rmtree(path)
+        shutil.rmtree(path, onerror=fixPermissions)
     except OSError as exception:
         if 'cannot call rmtree on a symbolic link' in str(exception).lower():
             os.unlink(path)

